### DISCUSSION TITLE
Does swc have the equivalent to tsconfig "noEmitOnError" : true ? How can we prevent compiling with errors?

### DISCUSSION THREAD

**Discussion by YarivGilad** [swc-project/swc#3063] [Q&A] [answered]
Does swc has the equivalent to tsconfig "noEmitOnError" : true ? 
How can we prevent compiling with errors?

**Comment by kdy1** [comment, marked as answer]
I think you mean type error (correct if me wrong, it's a guess based on the chat in discord) and as swc does not perform type checking, it's not possible.

**Reply by YarivGilad** [reply]
Thank you so much for your reply!
Indeed I'm referring to type errors 
If I understand correctly swc uses tsc for type-checking, right?
Is there anything I can config in tsconfig.json to fail the process if it contain typescript errors?
When I build my node.js TS project with swc, it seems trivial to prevent shipping bugs to production
That is why many teams use typescript to begin with - in order to catch bugs at compile time and prevent runtime errors as much as possible. so It seems critical to me to have a setup that doesn't finish the compilation if there are any TS errors... is there a way?

**Reply by kwonoj** [reply]
> If I understand correctly swc uses tsc for type-checking, right?

No, swc does not offer type checking. You should use typescript compiler separately to run type checks.

**Reply by YarivGilad** [reply]
Oh... okay... I don't know why but somehow I got the idea that swc is using tsc on its own for type=checking like esbuild does...
I guess I was wrong...
So just to sum it up, in case someone's reading it and needs a practical bottom line:
we could write an npm script that does something like this:    

`tsc --noEmit && swc src -d dist`

Thanks!

**Reply by kdy1** [reply]
I don't think esbuild will check types

**Reply by Smrtnyk** [reply]
esbuild definitely doesn't check types

**Reply by YarivGilad** [reply]
true. much like swc, esbuild doesn't have type checking and you need to run `tsc --noEmit` before running it on production builds
https://esbuild.github.io/content-types/#typescript
Thank you! this was an eye-opener :)
